x-pack/lib/monitoring/inputs/metrics.rb

```diff
@@ -105,15 +106,28 @@ def stop
     end
 
     def update(snapshot)
+      if LogStash::MonitoringExtension.use_direct_shipping?(LogStash::SETTINGS)
```

Comment:
In case direct shipping is enabled and a user change the ES output configuration in one of the pipelines those new `cluster_uuids` aren't brought up by the monitoring pipeline because the monitoring pipeline is not restarted.
Actually when the agent `converge_state_and_update` it reloads the changed pipelines and `update_metrics` but doesn't reload also the monitoring pipeline, should we scatter a refresh of that pipeline also?

Reply:
However this behaveior is almost aligned to what it's done for not direct shipping, because in that case the ES to ship data is loaded from `logstash.yml` and that ES is responsible to enrich the data with `cluster_uuid`, so that configuration to be changed force the user to restart Logstash

Reply:
I think I'm ok with this, and we can always follow up if this becomes an issue.

Its slightly different from the non-direct shipping case, as the `logstash.yml` changes would be a change to the location of the monitoring data, whereas the changes drawn from reloaded plugins would be a change to where that data is shown in the monitoring cluster.